### PR TITLE
Add embree3

### PR DESCRIPTION
**Pull request by joaander**
[Fresnel](https://github.com/conda-forge/fresnel-feedstock), a library I develop and conda-forge package that I maintain, will switch from embree v2 to embree v3 in its next release. Thus, I submit this embree3 package for inclusion in conda-forge.

The embree3 API is **not backwards compatible** with embree2. Upstream developers have taken steps to ensure that the two versions may be installed side by side. Specifically, headers are in include/embree2/ and include/embree3, and the new library is named libembree3. I have structured this recipe so that there should be no file collisions with the existing conda-forge [embree v2 conda-forge package](https://github.com/conda-forge/embree-feedstock). Specifically, I remove the tutorial binaries and example model files which are not needed for applications that link to embree anyways.

### CODE REVIEW

**Comment by joaander** [comment]
@conda-forge/staged-recipes embree3 is ready to review.

File: recipes/embree3/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  host:
+    - tbb
+
+  run:
+    - tbb
+
+test:
+  commands:
+    - test -f "${PREFIX}/lib/libembree3.so"  # [linux]
+    - test -f "${PREFIX}/lib/libembree3.dylib"  # [osx]
```
Comment (ocefpaf) [inline recipes/embree3/meta.yaml]: Not a blocker to merge but you can reduce these two lines to one with:

```yaml
- test -f "${PREFIX}/lib/libembree3.${SHLIB_EXT}"  # [not win]
```

File: recipes/embree3/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+  host:
+    - tbb
+
+  run:
+    - tbb
+
+test:
+  commands:
+    - test -f "${PREFIX}/lib/libembree3.${SHLIB_EXT}"  # [not win]
```
Comment (scopatz) [inline recipes/embree3/meta.yaml]: I think you don't need the `.` here because `${SHLIB_EXT}` starts with a `.`

Comment (joaander) [reply recipes/embree3/meta.yaml]: Thanks, that fixed the failing builds.

**Pull request merged**